python/ray/autoscaler/_private/aws/config.py

```diff
@@ -590,16 +590,20 @@ def _check_ami(config):
 
     region = config["provider"]["region"]
     default_ami = DEFAULT_AMI.get(region)
-    if not default_ami:
-        # If we do not provide a default AMI for the given region, noop.
-        return
 
     for key, node_type in config["available_node_types"].items():
         node_config = node_type["node_config"]
         node_ami = node_config.get("ImageId", "").lower()
         if node_ami in ["", "latest_dlami"]:
-            node_config["ImageId"] = default_ami
-            ami_src_info[key] = "dlami"
+            if not default_ami:
+                cli_logger.abort(
+                    f"Node type `{key}` does not have an ImageId and no default "
```

Comment:
To slightly reduce cognitive load, let's express in this message that `ImageId` is a subfield of  `nodeConfig`.